app/Judite/Models/Student.php

```diff
@@ -112,11 +129,43 @@ public function isEnrolledInCourse(Course $course): bool
      *
      * @param \App\Judite\Models\Course $course
      *
+     * @throws \App\Exceptions\StudentIsNotEnrolledInCourseException|\App\Exceptions\EnrollmentCannotBeDeleted
+     *
      * @return bool
      */
     public function unenroll(Course $course): bool
     {
-        return $this->enrollments()->where('course_id', $course->id)->delete();
+        if (! $this->isEnrolledInCourse($course)) {
+            throw new StudentIsNotEnrolledInCourseException($course);
+        }
+
+        $enrollment = $this->getEnrollmentWithShiftByCourse($course);
+
+        if (is_null($enrollment)) {
+            return $this->enrollments()->whereCourseId($course->id)->delete();
+        } else {
+            throw new EnrollmentCannotBeDeleted($enrollment, 'The enrollment cannot be deleted because it as an associated shift.');
+        }
+    }
+
+    /**
+     * Check if the enrollment of this student in the given course is deletable.
+     *
+     * @param \App\Judite\Models\Course $course
+     *
+     * @throws \App\Exceptions\StudentIsNotEnrolledInCourseException|\App\Exceptions\EnrollmentCannotBeDeleted
+     *
+     * @return bool
+     */
+    public function enrollmentIsDeletable(Course $course): bool
+    {
+        $enrollment = $this->getEnrollmentWithShiftByCourse($course);
+
+        if (is_null($enrollment) && app('settings')->withinExchangePeriod()) {
```

Comment:
Can't we condense this into a single line of `return (is_null($enrollment) && app('settings')->withinExchangePeriod())`? 
Or is this bad practice because of readability @fntneves?

Reply:
It can be done that way 👍